static/app/views/issueDetails/groupTagValues.tsx

```diff
@@ -43,55 +37,109 @@ type RouteParams = {
 type Props = {
   baseUrl: string;
   group: Group;
-  organization: Organization;
   environments?: string[];
   project?: Project;
-} & RouteComponentProps<RouteParams, {}>;
-
-type State = {
-  tag: Tag | null;
-  tagValueList: TagValue[] | null;
-  tagValueListPageLinks: string;
 };
 
 const DEFAULT_SORT = 'count';
 
-class GroupTagValues extends DeprecatedAsyncComponent<
-  Props & DeprecatedAsyncComponent['props'] & WithRouterProps,
-  State & DeprecatedAsyncComponent['state']
-> {
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {environments: environment, organization} = this.props;
-    const {groupId, tagKey} = this.props.params;
-    return [
-      ['tag', `/organizations/${organization.slug}/issues/${groupId}/tags/${tagKey}/`],
-      [
-        'tagValueList',
-        `/organizations/${organization.slug}/issues/${groupId}/tags/${tagKey}/values/`,
-        {query: {environment, sort: this.getSort()}},
-      ],
-    ];
-  }
-
-  getSort(): string {
-    return this.props.location.query.sort || DEFAULT_SORT;
-  }
+function useTagQueries({
+  group,
+  tagKey,
+  environments,
+  sort,
+}: {
+  group: Group;
+  sort: string | string[];
+  environments?: string[];
+  tagKey?: string;
+}) {
+  const organization = useOrganization();
+
+  const {
+    data: tagValueList,
+    isLoading: tagValueListIsLoading,
+    isError: tagValueListIsError,
+    getResponseHeader,
+  } = useApiQuery<TagValue[]>(
+    [
+      `/organizations/${organization.slug}/issues/${group.id}/tags/${tagKey}/values/`,
```

Comment:
I just noticed this, but shouldn't tagKey be required?

Reply:
You're right, I actually just changed the types to require a tagKey